src/zope/app/i18n/browser/synchronize.py

```diff
@@ -38,98 +55,90 @@ class Synchronize(BaseView):
                      _('Newer Local'), _('Does not exist')]
 
     def __init__(self, context, request):
-        super(Synchronize, self).__init__(context, request)
+        self.context = context
+        self.request = request
 
         self.sync_url = self.request.cookies.get(
             'sync_url', DEFAULT)
-        self.sync_url = urllib.unquote(self.sync_url)
+        self.sync_url = unquote(self.sync_url)
         self.sync_username = self.request.cookies.get('sync_username', 'admin')
         self.sync_password = self.request.cookies.get('sync_password', 'admin')
         self.sync_languages = filter(None, self.request.cookies.get(
             'sync_languages', '').split(','))
+        self._connection = None
+
+    def _make_sync_url(self):
+        # make sure the URL contains the http:// prefix
+        url = self.sync_url
+        if not url.startswith(('http://', 'https://')):
+            url = 'http://' + url
+
+        # Add username and password to the url.
+        parts = urlparse(url)
+        if '@' not in parts.netloc:
+            parts = list(parts)
+            parts[1] = self.sync_username + ':' + self.sync_password + '@' + parts[1]
```

Comment:
I guess you could? Even though `_replace` is technically documented I don't like using it.

Although the docs for `urlparse` only say

> The return value is actually an instance of a subclass of tuple

So it isn't required to actually be a namedtuple.